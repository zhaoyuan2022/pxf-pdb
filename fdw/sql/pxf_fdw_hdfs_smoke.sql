-- data prep
-- start_ignore
{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} mkdir -p {{ TEST_LOCATION }}
\!mkdir -p {{ TEST_LOCATION }}
COPY (
	SELECT format('row_%s',i::varchar(255)),
		i,
		i+0.0001,
		i*100000000000,
		CASE WHEN (i%2) = 0 THEN 'true' ELSE 'false' END
		from generate_series(1, 100) s(i)
	) TO '{{ TEST_LOCATION }}/data.csv'
	WITH (FORMAT 'csv');
{{ GPDB_REMOTE }}-- if GPDB is remote, will need to scp file down from there for beeline
{{ GPDB_REMOTE }}\!scp {{ PGHOST }}:{{ TEST_LOCATION }}/data.csv {{ TEST_LOCATION }}
\!{{ HCFS_CMD }} dfs -mkdir -p '{{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}'
\!{{ HCFS_CMD }} dfs -copyFromLocal '{{ TEST_LOCATION }}/data.csv' '{{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}'
-- end_ignore

-- FDW test
CREATE SERVER hdfs_smoke_test_hdfs_server
	FOREIGN DATA WRAPPER {{ HCFS_PROTOCOL }}_pxf_fdw
	OPTIONS (config '{{ SERVER_CONFIG }}');
CREATE USER MAPPING FOR CURRENT_USER SERVER hdfs_smoke_test_hdfs_server;
CREATE FOREIGN TABLE hdfs_smoke_test_foreign_table (
		name TEXT,
		num INTEGER,
		dub DOUBLE PRECISION,
		longNum BIGINT,
		bool BOOLEAN
	) SERVER hdfs_smoke_test_hdfs_server
	OPTIONS (resource '{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/data.csv', format 'csv');
SELECT * FROM hdfs_smoke_test_foreign_table ORDER BY name;
SELECT name, num FROM hdfs_smoke_test_foreign_table WHERE num > 50 ORDER BY name;

-- start_ignore
{{ CLEAN_UP }}-- clean up HCFS and local disk
{{ CLEAN_UP }}\!{{ HCFS_CMD }} dfs -rm -r -f {{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}
{{ CLEAN_UP }}\!rm -rf {{ TEST_LOCATION }}
{{ CLEAN_UP }}{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} rm -rf {{ TEST_LOCATION }}
-- end_ignore
