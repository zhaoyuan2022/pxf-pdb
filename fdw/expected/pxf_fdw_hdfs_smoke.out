-- data prep
-- start_ignore
{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} mkdir -p {{ TEST_LOCATION }}
\!mkdir -p {{ TEST_LOCATION }}
COPY (
	SELECT format('row_%s',i::varchar(255)),
		i,
		i+0.0001,
		i*100000000000,
		CASE WHEN (i%2) = 0 THEN 'true' ELSE 'false' END
		from generate_series(1, 100) s(i)
	) TO '{{ TEST_LOCATION }}/data.csv'
	WITH (FORMAT 'csv');
{{ GPDB_REMOTE }}-- if GPDB is remote, will need to scp file down from there for beeline
{{ GPDB_REMOTE }}\!scp {{ PGHOST }}:{{ TEST_LOCATION }}/data.csv {{ TEST_LOCATION }}
\!{{ HCFS_CMD }} dfs -mkdir -p '{{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}'
\!{{ HCFS_CMD }} dfs -copyFromLocal '{{ TEST_LOCATION }}/data.csv' '{{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}'
-- end_ignore
-- FDW test
CREATE SERVER hdfs_smoke_test_hdfs_server
	FOREIGN DATA WRAPPER {{ HCFS_PROTOCOL }}_pxf_fdw
	OPTIONS (config '{{ SERVER_CONFIG }}');
CREATE USER MAPPING FOR CURRENT_USER SERVER hdfs_smoke_test_hdfs_server;
CREATE FOREIGN TABLE hdfs_smoke_test_foreign_table (
		name TEXT,
		num INTEGER,
		dub DOUBLE PRECISION,
		longNum BIGINT,
		bool BOOLEAN
	) SERVER hdfs_smoke_test_hdfs_server
	OPTIONS (resource '{{ HCFS_BUCKET }}{{ TEST_LOCATION }}/data.csv', format 'csv');
SELECT * FROM hdfs_smoke_test_foreign_table ORDER BY name;
  name   | num |   dub    |    longnum     | bool 
---------+-----+----------+----------------+------
 row_1   |   1 |   1.0001 |   100000000000 | f
 row_10  |  10 |  10.0001 |  1000000000000 | t
 row_100 | 100 | 100.0001 | 10000000000000 | t
 row_11  |  11 |  11.0001 |  1100000000000 | f
 row_12  |  12 |  12.0001 |  1200000000000 | t
 row_13  |  13 |  13.0001 |  1300000000000 | f
 row_14  |  14 |  14.0001 |  1400000000000 | t
 row_15  |  15 |  15.0001 |  1500000000000 | f
 row_16  |  16 |  16.0001 |  1600000000000 | t
 row_17  |  17 |  17.0001 |  1700000000000 | f
 row_18  |  18 |  18.0001 |  1800000000000 | t
 row_19  |  19 |  19.0001 |  1900000000000 | f
 row_2   |   2 |   2.0001 |   200000000000 | t
 row_20  |  20 |  20.0001 |  2000000000000 | t
 row_21  |  21 |  21.0001 |  2100000000000 | f
 row_22  |  22 |  22.0001 |  2200000000000 | t
 row_23  |  23 |  23.0001 |  2300000000000 | f
 row_24  |  24 |  24.0001 |  2400000000000 | t
 row_25  |  25 |  25.0001 |  2500000000000 | f
 row_26  |  26 |  26.0001 |  2600000000000 | t
 row_27  |  27 |  27.0001 |  2700000000000 | f
 row_28  |  28 |  28.0001 |  2800000000000 | t
 row_29  |  29 |  29.0001 |  2900000000000 | f
 row_3   |   3 |   3.0001 |   300000000000 | f
 row_30  |  30 |  30.0001 |  3000000000000 | t
 row_31  |  31 |  31.0001 |  3100000000000 | f
 row_32  |  32 |  32.0001 |  3200000000000 | t
 row_33  |  33 |  33.0001 |  3300000000000 | f
 row_34  |  34 |  34.0001 |  3400000000000 | t
 row_35  |  35 |  35.0001 |  3500000000000 | f
 row_36  |  36 |  36.0001 |  3600000000000 | t
 row_37  |  37 |  37.0001 |  3700000000000 | f
 row_38  |  38 |  38.0001 |  3800000000000 | t
 row_39  |  39 |  39.0001 |  3900000000000 | f
 row_4   |   4 |   4.0001 |   400000000000 | t
 row_40  |  40 |  40.0001 |  4000000000000 | t
 row_41  |  41 |  41.0001 |  4100000000000 | f
 row_42  |  42 |  42.0001 |  4200000000000 | t
 row_43  |  43 |  43.0001 |  4300000000000 | f
 row_44  |  44 |  44.0001 |  4400000000000 | t
 row_45  |  45 |  45.0001 |  4500000000000 | f
 row_46  |  46 |  46.0001 |  4600000000000 | t
 row_47  |  47 |  47.0001 |  4700000000000 | f
 row_48  |  48 |  48.0001 |  4800000000000 | t
 row_49  |  49 |  49.0001 |  4900000000000 | f
 row_5   |   5 |   5.0001 |   500000000000 | f
 row_50  |  50 |  50.0001 |  5000000000000 | t
 row_51  |  51 |  51.0001 |  5100000000000 | f
 row_52  |  52 |  52.0001 |  5200000000000 | t
 row_53  |  53 |  53.0001 |  5300000000000 | f
 row_54  |  54 |  54.0001 |  5400000000000 | t
 row_55  |  55 |  55.0001 |  5500000000000 | f
 row_56  |  56 |  56.0001 |  5600000000000 | t
 row_57  |  57 |  57.0001 |  5700000000000 | f
 row_58  |  58 |  58.0001 |  5800000000000 | t
 row_59  |  59 |  59.0001 |  5900000000000 | f
 row_6   |   6 |   6.0001 |   600000000000 | t
 row_60  |  60 |  60.0001 |  6000000000000 | t
 row_61  |  61 |  61.0001 |  6100000000000 | f
 row_62  |  62 |  62.0001 |  6200000000000 | t
 row_63  |  63 |  63.0001 |  6300000000000 | f
 row_64  |  64 |  64.0001 |  6400000000000 | t
 row_65  |  65 |  65.0001 |  6500000000000 | f
 row_66  |  66 |  66.0001 |  6600000000000 | t
 row_67  |  67 |  67.0001 |  6700000000000 | f
 row_68  |  68 |  68.0001 |  6800000000000 | t
 row_69  |  69 |  69.0001 |  6900000000000 | f
 row_7   |   7 |   7.0001 |   700000000000 | f
 row_70  |  70 |  70.0001 |  7000000000000 | t
 row_71  |  71 |  71.0001 |  7100000000000 | f
 row_72  |  72 |  72.0001 |  7200000000000 | t
 row_73  |  73 |  73.0001 |  7300000000000 | f
 row_74  |  74 |  74.0001 |  7400000000000 | t
 row_75  |  75 |  75.0001 |  7500000000000 | f
 row_76  |  76 |  76.0001 |  7600000000000 | t
 row_77  |  77 |  77.0001 |  7700000000000 | f
 row_78  |  78 |  78.0001 |  7800000000000 | t
 row_79  |  79 |  79.0001 |  7900000000000 | f
 row_8   |   8 |   8.0001 |   800000000000 | t
 row_80  |  80 |  80.0001 |  8000000000000 | t
 row_81  |  81 |  81.0001 |  8100000000000 | f
 row_82  |  82 |  82.0001 |  8200000000000 | t
 row_83  |  83 |  83.0001 |  8300000000000 | f
 row_84  |  84 |  84.0001 |  8400000000000 | t
 row_85  |  85 |  85.0001 |  8500000000000 | f
 row_86  |  86 |  86.0001 |  8600000000000 | t
 row_87  |  87 |  87.0001 |  8700000000000 | f
 row_88  |  88 |  88.0001 |  8800000000000 | t
 row_89  |  89 |  89.0001 |  8900000000000 | f
 row_9   |   9 |   9.0001 |   900000000000 | f
 row_90  |  90 |  90.0001 |  9000000000000 | t
 row_91  |  91 |  91.0001 |  9100000000000 | f
 row_92  |  92 |  92.0001 |  9200000000000 | t
 row_93  |  93 |  93.0001 |  9300000000000 | f
 row_94  |  94 |  94.0001 |  9400000000000 | t
 row_95  |  95 |  95.0001 |  9500000000000 | f
 row_96  |  96 |  96.0001 |  9600000000000 | t
 row_97  |  97 |  97.0001 |  9700000000000 | f
 row_98  |  98 |  98.0001 |  9800000000000 | t
 row_99  |  99 |  99.0001 |  9900000000000 | f
(100 rows)

SELECT name, num FROM hdfs_smoke_test_foreign_table WHERE num > 50 ORDER BY name;
  name   | num 
---------+-----
 row_100 | 100
 row_51  |  51
 row_52  |  52
 row_53  |  53
 row_54  |  54
 row_55  |  55
 row_56  |  56
 row_57  |  57
 row_58  |  58
 row_59  |  59
 row_60  |  60
 row_61  |  61
 row_62  |  62
 row_63  |  63
 row_64  |  64
 row_65  |  65
 row_66  |  66
 row_67  |  67
 row_68  |  68
 row_69  |  69
 row_70  |  70
 row_71  |  71
 row_72  |  72
 row_73  |  73
 row_74  |  74
 row_75  |  75
 row_76  |  76
 row_77  |  77
 row_78  |  78
 row_79  |  79
 row_80  |  80
 row_81  |  81
 row_82  |  82
 row_83  |  83
 row_84  |  84
 row_85  |  85
 row_86  |  86
 row_87  |  87
 row_88  |  88
 row_89  |  89
 row_90  |  90
 row_91  |  91
 row_92  |  92
 row_93  |  93
 row_94  |  94
 row_95  |  95
 row_96  |  96
 row_97  |  97
 row_98  |  98
 row_99  |  99
(50 rows)

-- start_ignore
{{ CLEAN_UP }}-- clean up HCFS and local disk
{{ CLEAN_UP }}\!{{ HCFS_CMD }} dfs -rm -r -f {{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}
{{ CLEAN_UP }}Deleted {{ HCFS_SCHEME }}{{ HCFS_BUCKET }}{{ TEST_LOCATION }}
{{ CLEAN_UP }}\!rm -rf {{ TEST_LOCATION }}
{{ CLEAN_UP }}{{ GPDB_REMOTE }}\!ssh {{ PGHOST }} rm -rf {{ TEST_LOCATION }}
-- end_ignore
